src/user/library/index.ts

```diff
@@ -0,0 +1,209 @@
+import {AdSetForm, CampaignForm, Creative, OS, Segment} from "../views/adsManager/types";
+import {
+  CreateAdInput,
+  CreateAdSetInput,
+  CreateCampaignInput,
+  CreativeInput, GeocodeInput, UpdateAdSetInput, UpdateCampaignInput
+} from "../../graphql/types";
+import axios from "axios";
+import {print} from "graphql";
+import {CreateCreativeDocument} from "../../graphql/creative.generated";
+import {CampaignFragment} from "../../graphql/campaign.generated";
+import {CreateAdDocument} from "../../graphql/ad-set.generated";
+
+export async function transformNewForm(
+  form: CampaignForm,
+  auth: any,
+  advertiserId: string
+): Promise<CreateCampaignInput> {
+  const adSets = form.adSets;
+  const transformedAdSet: CreateAdSetInput[] = [];
+
+  for (let adSet of adSets) {
+    const ads: CreateAdInput[] = [];
+
+    for (let ad of adSet.creatives) {
+      const creative = await transformCreative(
+        ad,
+        adSet,
+        auth.accessToken,
+        advertiserId
+      );
+      ads.push(creative);
+    }
```

Comment:
What is happening here is that I nee to create the creative _before_ the campaign.